lingua_franca/lang/format_uk.py

```diff
@@ -400,25 +396,26 @@ def nice_time_uk(dt, speech=True, use_24hour=True, use_ampm=False):
         else:
             speak = pronounce_hour_uk(hour)
 
+            if use_ampm:
+                if dt.hour < 4:
+                    speak += " ночі"
+                elif dt.hour < 12:
+                    speak += " ранку"
+                elif dt.hour < 18:
+                    speak += " дня"
+                else:
+                    speak += " вечора"
+
             if dt.minute == 0:
                 if not use_ampm:
                     if dt.hour % 12 == 1:
                         return speak
-                    return speak + " " + plural_uk(dt.hour % 12, "година", "години", "годин")
+                    return speak + " " + plural_uk(dt.hour % 12, "година", "години", "годин", "годиною", "годинами", "годині")
```

Comment:
i guess you want to replace the positionals (`one`, `few`, `many`)

Reply:
@NeonDmitry Is this something you can address?